spec/src/main/asciidoc/app-programming-model.adoc

```diff
@@ -119,8 +119,7 @@ The following Annotations exist, see below for common fields:
 |@ConcurrentGauge | M, C, T | Denotes a gauge which counts the parallel invocations of the annotated object. | MetricUnits.NONE
 |@Gauge   | M | Denotes a gauge, which samples the value of the annotated object.  | _no default_, must be supplied by the user
 |@Metered | M, C, T  | Denotes a meter, which tracks the frequency of invocations of the annotated object. | MetricUnits.PER_SECOND
-|@Metric  | M, F, P | An annotation that contains the metadata information when requesting a metric to be injected or produced. This annotation can be used on fields
-    of type `Meter`, `Timer`, `Counter`, and `Histogram`. For `Gauge`, the `@Metric` annotation can only be used on producer methods/fields. | MetricUnits.NONE
+|@Metric  | F, P | An annotation that contains the metadata information when requesting a metric to be injected. | MetricUnits.NONE
```

Comment:
I removed the M (method) target here but kept it on the actual annotation type (`Metric.java`) in case that an implementation chooses to keep supporting this, this way we don't prevent them from doing that.

Reply:
@Inject is fine. @Produces is not. ;) So I’m otherwise ok with Method/Field.

Reply:
Yep. @Inject is fine.

Reply:
Keeping the M (method) target on the annotation, and having that be out of sync with the spec, seems a bit easter-eggy.  Shouldn't we just remove it from the annotation?

Reply:
I don't know - on the other hand some implementations may still choose to support it for whatever reason and we don't want to break them... When it comes to breaking changes, I tend to say less is better.
Anyone else has an opinion on this?

Reply:
It may be necessary to remove the method target off the annotation, from a "closing down loopholes" perspective.

Yes, it means implementations can't support the previous behavior, but it also prevents it from being used in a way the spec doesn't intend. And thus requiring implementations to warn or otherwise fail when it is used on a method

Reply:
Ok I removed it from the annotation now.